### ISSUE TITLE
Question: Disabling private error handlers to conserve RAM

### ISSUE THREAD

**Issue by shreyasbharath** [ETLCPP/etl#270]
Hello library maintainers,

Thanks for an awesome library for us embedded C++ folks.

I am noticing a lot of private error handlers being defined in anything that explicitly or implicitly #includes `etl/delegate.h` in the target map (ARM GCC) file. I have defined `ETL_NO_CHECKS` so I'd expect to pay no cost in terms of RAM. I see hundreds of instances of these in the map file, which I feel is wasting precious RAM that can be used for some other purpose.

```
_ZGVN3etl21private_error_handler7wrapperIvE10invocationE (4 bytes)
_ZN3etl21private_error_handler7wrapperIvE10invocationE (8 bytes)
```

I suspect these are being defined by this class in `error_handler.h`

```cpp
namespace etl
{
    namespace private_error_handler
    {
        // A wrapper template to allow static definition in header.
        template <typename TDummy> struct wrapper
        {
            typedef void (*stub_type)(void *object, const etl::exception &);

            //*************************************************************************
            /// The internal invocation object.
            //*************************************************************************
            struct invocation_element
            {
                //***********************************************************************
                invocation_element() : object(ETL_NULLPTR), stub(ETL_NULLPTR)
                {
                }

                //***********************************************************************
                void *object;
                stub_type stub;
            };

            static invocation_element invocation;		// I presume this is what's consuming RAM?
        };
...
```

How do I compile these away from my final binary?

**Comment by jwellbelove** [comment]
Thanks for the heads up on that.
I probably need to add some conditional compilation macros around that.
I'll take a look this evening.

**Comment by jwellbelove** [comment]
Fixed in 18.15.4

**Comment by shreyasbharath** [comment]
Thank you!
